contracts/shapella-upgrade/ShapellaUpgradeTemplate.sol

```diff
@@ -0,0 +1,870 @@
+    IHashConsensus public constant _hashConsensusForValidatorsExitBusOracle = IHashConsensus(0x8374B4aC337D7e367Ea1eF54bB29880C3f036A51);
+    address public constant _eip712StETH = 0xB4300103FfD326f77FfB3CA54248099Fb29C3b9e;
+    address public constant _voting = 0xbc0B67b4553f4CF52a913DE9A6eD0057E2E758Db;
+    address public constant _agent = 0x4333218072D5d7008546737786663c38B4D561A4;
+    INodeOperatorsRegistry public constant _nodeOperatorsRegistry = INodeOperatorsRegistry(0x9D4AF1Ee19Dad8857db3a45B0374c81c8A1C6320);
+    address public constant _gateSeal = 0x75A77AE52d88999D0b12C6e5fABB1C1ef7E92638;
+    address public constant _withdrawalQueueImplementation = 0xF7a378BB9E911550baA5e729f5Ab1592aDD905A5;
+    address public constant _stakingRouterImplementation = 0xb02791097DE4B7B83265C9516640C8223830a351;
+    address public constant _accountingOracleImplementation = 0x49cc40EE660BfD5f46423f04891502410d32E965;
+    address public constant _validatorsExitBusOracleImplementation = 0xBE378f865Ab69f51d8874aeB9508cbbC42B3FBDE;
+    address public constant _dummyImplementation = 0x6A03b1BbB79460169a205eFBCBc77ebE1011bCf8;
+    address public constant _locatorImplementation = 0x6D5b7439c166A1BDc5c8DB547c1a871c082CE22C;
+    address public constant _withdrawalVaultImplementation = 0x297Eb629655C8c488Eb26442cF4dfC8A7Cc32fFb;
+    address public constant _previousDepositSecurityModule = 0x7DC1C1ff64078f73C98338e2f17D1996ffBb2eDe;
+
+    IAccountingOracle public constant _accountingOracle = IAccountingOracle(0x76f358A842defa0E179a8970767CFf668Fc134d6);
+    IBurner public constant _burner = IBurner(0x20c61C07C2E2FAb04BF5b4E12ce45a459a18f3B1);
+    IDepositSecurityModule public constant _depositSecurityModule = IDepositSecurityModule(0xC8a75E7196b11aE2DEbC39a2F8583f852E5BB7c3);
+    ILido public constant _lido = ILido(0x1643E812aE58766192Cf7D2Cf9567dF2C37e9B7F);
+    ILidoOracle public constant _lidoOracle = ILidoOracle(0x24d8451BC07e7aF4Ba94F69aCDD9ad3c6579D9FB);
+    // _legacyOracle has the same address as _lidoOracle: we're renaming the contract, but it's on the same address
+    ILegacyOracle public constant _legacyOracle = ILegacyOracle(address(_lidoOracle));
+    IOracleDaemonConfig public constant _oracleDaemonConfig = IOracleDaemonConfig(0xad55833Dec7ab353B47691e58779Bd979d459388);
+    IOracleReportSanityChecker public constant _oracleReportSanityChecker = IOracleReportSanityChecker(0x0F3475f755FA356f1356ABC80B4aE4a786d8aae5);
+    IStakingRouter public constant _stakingRouter = IStakingRouter(0xa3Dbd317E53D363176359E10948BA0b1c0A4c820);
+    IValidatorsExitBusOracle public constant _validatorsExitBusOracle = IValidatorsExitBusOracle(0xb75A55EFab5A8f5224Ae93B34B25741EDd3da98b);
+    IWithdrawalQueue public constant _withdrawalQueue = IWithdrawalQueue(0xCF117961421cA9e546cD7f50bC73abCdB3039533);
+    IWithdrawalVault public constant _withdrawalVault = IWithdrawalVault(0xdc62f9e8C34be08501Cdef4EBDE0a280f576D762);
```

Comment:
Proceeding to mainnet, I would consider restructuring constants to have more visibility for:
- new proxies
- existing proxies
- new implementations
- new non-upgradable contracts
- old non-upgradable contracts

It's only a rough idea, maybe you could come up with something better